docs/source/docs/configuration.md

```diff
@@ -2,19 +2,515 @@
+
+- `extract_levels`: list of positive ints (ie, the min value in `extract_levels` should
+  be >=0). Eg. [1, 2, 3, 5] will extract information at those levels (but not 4).
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+```
+
+### Loss - required
+
+This section defines the loss in training.
+
+The losses in DeepReg are defined depending on the type of network to be built, and can
+be split into two sections: dissimilarity loss (between moving and fixed tensors), and
+regularization losses (on the DDFs predicted).
+
+DeepReg uses `tf.keras.Model` `add_loss()` in the Registry method to add losses to the
+model, which provides some flexibility in configuration.
+
+Currently, DeepReg offers conditional, ddf/dvf and affine registration pre-built models.
+The models are configured to add the losses as follows:
+
+- Conditional: label loss.
+- DDF/DVF: ddf loss, image loss, label loss.
+- Affine: ddf loss, image loss, label loss.
+
+Therefore, currently, the config file requires all the appropriate fields to be passed
+such that the model can be built accordingly.
+
+#### Image
+
+- `weight`: float type, weight of individual loss element in total loss function.
+- `name`: string type, one of "lncc", "ssd" or "gmi".
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    dissimilarity:
+      image:
+        name: "lncc" # other options include "lncc", "ssd" and "gmi", for local normalised cross correlation,
+        weight: 0.1
+```
+
+#### Label
+
+- `weight`: float type, weight of individual loss element in total loss function.
+- `name`: string type, "multi_scale" or "single_scale", defines the resolution of
+  kernels to use with the loss.
+  - The same string should be passed as a field to the dictionary, with the following
+    additional arguments under each field:
+  - `multi_scale`:
+    - `loss_type`: string type, name of loss to use. Options currently include "dice",
+      "cross-entropy", "mean-squared", "generalised_dice" and "jaccard".
+    - `loss_scales`: list of ints, what scales to output the loss
+  - `single_scale`:
+    - `loss_type`: name of loss to use. Options currently include "dice",
+      "cross-entropy", "mean-squared", "generalised_dice" and "jaccard".
+
+EG.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    dissimilarity:
```

Comment:
in the latest `main` this `dissimilarity` layer has been removed.

@NMontanaBrown if you want I can update this branch for you

Reply:
The only thing left is the multi-scales argument. So now, how do we do scales for losses? previously we had:

```yaml
loss:
    image:
      name: "gmi"
      weight: 1.0
    label:
      weight: 1.0
      name: "multi_scale" # "multi_scale" or "single_scale"
      multi_scale:
        loss_type: "dice" # options include "dice", "cross-entropy", "mean-squared", "generalised_dice" and "jaccard"
        loss_scales: [0, 1, 2, 4, 8, 16, 32]
```

now.... we don't specify the argument "multi-scale" or single scale, we just pass the name: "dice", then scales: a list, and if list len >1, it is understood as multiscale?

Reply:
so it would be:
```yaml
label:
      weight: 1.0
      name: "dice"
      scales: [0, 1, 2, 4, 8, 16, 32]
```
is interpreted as multi-scale dice
and then : 
```yaml
label:
      weight: 1.0
      name: "dice"
      scales: [1]
```
is interpreted as single scale dice?

Reply:
check https://github.com/DeepRegNet/DeepReg/blob/main/deepreg/model/loss/label.py#L42

So if you want to use single scale` just not to put `scales` at all, or leave empty, [1] works too

is it understandable? any suggestions?

Reply:
let me be more detailed, to be single scale, we can do

```yaml
label:
      weight: 1.0
      name: "dice"
      scales: [0]
```
important point, no scaling means value = 0

or

```yaml
label:
      weight: 1.0
      name: "dice"
      scales: null
```

or

```yaml
label:
      weight: 1.0
      name: "dice"
      scales: 
```

But we should not give an empty list (for now)

```yaml
label:
      weight: 1.0
      name: "dice"
      scales: []
```

Reply:
maybe I can update the code, to make `[]` -> single scale ^^, what do you think

Reply:
Updating, I just saw this. Add ticket for empty list to be single scale. I will include in the docs at the moment that that will break the code.